src/services/codefixes/inferFromUsage.ts

```diff
@@ -333,21 +333,50 @@ namespace ts.codefix {
     }
 
     function annotateJSDocParameters(changes: textChanges.ChangeTracker, sourceFile: SourceFile, parameterInferences: readonly ParameterInference[], program: Program, host: LanguageServiceHost): void {
-        const signature = parameterInferences.length && parameterInferences[0].declaration.parent;
+        const signature = parameterInferences.length && first(parameterInferences).declaration.parent;
         if (!signature) {
             return;
         }
-        const paramTags = mapDefined(parameterInferences, inference => {
+
+        const inferences = mapDefined(parameterInferences, inference => {
             const param = inference.declaration;
             // only infer parameters that have (1) no type and (2) an accessible inferred type
-            if (param.initializer || getJSDocType(param) || !isIdentifier(param.name)) return;
-
+            if (param.initializer || getJSDocType(param) || !isIdentifier(param.name)) {
+                return;
+            }
             const typeNode = inference.type && getTypeNodeIfAccessible(inference.type, param, program, host);
-            const name = factory.cloneNode(param.name);
-            setEmitFlags(name, EmitFlags.NoComments | EmitFlags.NoNestedComments);
-            return typeNode && factory.createJSDocParameterTag(/*tagName*/ undefined, name, /*isBracketed*/ !!inference.isOptional, factory.createJSDocTypeExpression(typeNode), /* isNameFirst */ false, "");
+            if (typeNode) {
+                const name = factory.cloneNode(param.name);
+                setEmitFlags(name, EmitFlags.NoComments | EmitFlags.NoNestedComments);
+                return { name: factory.cloneNode(param.name), param, isOptional: !!inference.isOptional, typeNode };
+            }
         });
-        addJSDocTags(changes, sourceFile, signature, paramTags);
+
+        if (!inferences.length) {
+            return;
+        }
+
+        if ((isArrowFunction(signature) || isFunctionExpression(signature)) && isCallExpression(signature.parent)) {
+            const needParens = isArrowFunction(signature) && !findChildOfKind(signature, SyntaxKind.OpenParenToken, sourceFile);
+            if (needParens) {
+                changes.insertNodeBefore(sourceFile, first(signature.parameters), factory.createToken(SyntaxKind.OpenParenToken));
+            }
+
+            forEach(inferences, ({ typeNode, param }) => {

```

Comment:
I'm not sure if it's better, but you could also generate:

```ts
return names.filter(/** 
 * @param {string} name
 * @param {number} index
 */ (string, index ) => ....)
```

That's probably what a human would write, but it's way harder to format properly.

Did you consider that option? Do you think it would work?
```

Reply:
No, I didn't. I just used the option from the `Expected behavior` section. If the proposed option better, I can try to change. I think we can ask the author of the issue if it is an acceptable option for him.

Maybe comment should be formatted like so

```ts
return names.filter(
    /**
     * @param {string} name
     * @param {number} index
     */
    (string, index) => index
);
```

Reply:
I commented on the bug, but let's proceed with what you have now.

Reply:
Oke